frontend/src/pages/Wallet.js

```diff
@@ -0,0 +1,143 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2021 
+
+import {
+	Button,
+	Table,
+	TableBody,
+	TableCell,
+	TableContainer,
+	TableRow,
+} from "@mui/material";
+import React, { useEffect } from "react";
+import appwriteApi from "../api/appwriteApi";
+import { useHistory } from "react-router-dom";
+import Grid from "@mui/material/Grid";
+import { useState } from "react";
+import detectEthereumProvider from "@metamask/detect-provider";
+import CenterFlexBoxMedium from "../components/CenterFlexBoxMedium";
+
+function AccountEntry({ data }) {
+	return 	<TableRow>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>{data.pubAddres}</TableCell>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>{parseInt(data.bl, 16)}</TableCell>
+	</TableRow>;
+}
+
+function EthAccountsDetail({ publicAddresses }) {
+	const [detailData, setDetailData] = useState([]);
+
+	useEffect(() => {
+		async function fetchAccountsDetails(publicAddresses) {
+			const ethereum = await detectEthereumProvider();
```

Comment:
Could you use a boolean-indicating name like `hasDectedEthereumProvider` instead of `ethereum`?

Reply:
no, because an Ethereum-wise object is returned if success

Reply:
I thought about an additional variable `const hasDectectedEthereumProvider =  (ethereum != null)` in between.

In this case, it's not dramatic, you could also inline it. But using an arbitrary variable as boolean value which without  a name that would indicate boolean meaning is not generally clear in my opinion.